relay-event-schema/src/protocol/contexts/trace.rs

```diff
@@ -10,19 +13,84 @@ use crate::protocol::{OperationType, OriginType, SpanData, SpanLink, SpanStatus}
 #[derive(Clone, Debug, Default, PartialEq, Empty, IntoValue, ProcessValue)]
 pub struct TraceId(pub String);
 
+impl TraceId {
+    /// Converts the trace ID to an u128 value.
+    ///
+    /// # Returns
+    ///
+    /// This method will return `u128::MIN` (0) if:
+    /// - The trace ID string is empty
+    /// - The trace ID string contains invalid hex characters
+    /// - The trace ID string is not exactly 32 characters
+    ///
+    /// Otherwise, it returns the trace ID as an u128 integer.
+    pub fn as_u128(&self) -> u128 {
```

Comment:
This conversion assumes that for each byte we can take out a hex value from `0-15`. This is not super efficient because we could technically represent the same data with 16 bytes instead of 32, but for the sake of simplicity (given the `Empty`, `IntoValue` and `ProcessValue`), we will keep the type as a `String`. We might want to make this more efficient in the future.

Reply:
As discussed, probably worth switching all of this now to `[u8; 16]` like `EventId`